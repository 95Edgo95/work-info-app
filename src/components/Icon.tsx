import classNames from "classnames";
import * as React from "react";

import * as styles from "assets/css/icon.css";

import "assets/images/svgSprite.icon.svg";

interface IIconProps {
  name: string;
  size?: number;
  className?: string;
  color?: string;
  style?: object;

  [key: string]: any;
}

export default function Icon({ name, size = 24, className = '', color = 'black', style = {}, ...props }: IIconProps) {
  const iconWrapperClasses = classNames({
    [styles.iconWrapper]: true,
    [className]: true,
  });

  return (
    <span
      className={iconWrapperClasses}
      style={{
        fontSize: `${size}px`,
        color,
        ...style,
      }}
      role='presentation'
      {...props}
    >
      <svg className={styles.icon}>
        <use xlinkHref={`#${name}`}/>
      </svg>
    </span>
  );
}
