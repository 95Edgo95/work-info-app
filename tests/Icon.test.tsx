import Icon from "../src/components/Icon";
import { mount } from "enzyme";
import * as React from "react";

describe('Icon', () => {
  it('should render correctly', () => {
    const component = mount(
      <Icon
        name="pencil"
      />
    );

    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it('should render correctly with size', () => {
    const component = mount(
      <Icon
        name="pencil"
        size={32}
      />
    );

    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it('should render correctly with color', () => {
    const component = mount(
      <Icon
        name="pencil"
        color="red"
        size={32}
      />
    );

    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it('should render correctly with style', () => {
    const component = mount(
      <Icon
        style={{ margin: "0 auto" }}
        name="pencil"
        color="red"
        size={32}
      />
    );

    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
