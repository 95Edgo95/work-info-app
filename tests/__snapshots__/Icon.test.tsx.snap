// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render correctly 1`] = `
<Icon
  name="pencil"
>
  <span
    className="undefined "
    role="presentation"
    style={
      Object {
        "color": "black",
        "fontSize": "24px",
      }
    }
  >
    <svg>
      <use
        xlinkHref="#pencil"
      />
    </svg>
  </span>
</Icon>
`;

exports[`Icon should render correctly with color 1`] = `
<Icon
  color="red"
  name="pencil"
  size={32}
>
  <span
    className="undefined "
    role="presentation"
    style={
      Object {
        "color": "red",
        "fontSize": "32px",
      }
    }
  >
    <svg>
      <use
        xlinkHref="#pencil"
      />
    </svg>
  </span>
</Icon>
`;

exports[`Icon should render correctly with size 1`] = `
<Icon
  name="pencil"
  size={32}
>
  <span
    className="undefined "
    role="presentation"
    style={
      Object {
        "color": "black",
        "fontSize": "32px",
      }
    }
  >
    <svg>
      <use
        xlinkHref="#pencil"
      />
    </svg>
  </span>
</Icon>
`;

exports[`Icon should render correctly with style 1`] = `
<Icon
  color="red"
  name="pencil"
  size={32}
  style={
    Object {
      "margin": "0 auto",
    }
  }
>
  <span
    className="undefined "
    role="presentation"
    style={
      Object {
        "color": "red",
        "fontSize": "32px",
        "margin": "0 auto",
      }
    }
  >
    <svg>
      <use
        xlinkHref="#pencil"
      />
    </svg>
  </span>
</Icon>
`;
